app/src/main/java/com/xinto/opencord/db/dao/ChannelsDao.kt

```diff
@@ -0,0 +1,34 @@
+package com.xinto.opencord.db.dao
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import com.xinto.opencord.db.entity.channel.EntityChannel
+
+@Dao
+interface ChannelsDao {
+    // --------------- Inserts ---------------
+    @Insert(
+        onConflict = OnConflictStrategy.REPLACE,
+        entity = EntityChannel::class,
+    )
+    fun insertChannels(channels: List<EntityChannel>)
```

Comment:
mark these DAO functions as suspendable and then call them from the coroutine

Reply:
The issue with this is that transactions won't work properly anymore

Here for example, it's already in a coroutine from the listener, however then it gets cancelled out by the transaction.
putting the calls already inside the transaction to a coroutine ruins the point of them
https://github.com/MateriiApps/OpenCord/blob/dc380c7eb30ee25de97f7cb6111da45f3e2fb51c/app/src/main/java/com/xinto/opencord/store/MessageStore.kt#L166-L177

Reply:
then why put them in transaction? isn't transaction supposed to be "async for java"

Reply:
No transaction are for optimizing db calls, not async

Reply:
guhhh okay

Reply:
but like with coroutines you don't even need the transaction block

Reply:
The point of coroutines ≠ transactions
it's not just sync vs async